[14.0][FIX] purchase_discount: Prevent price_unit recursion error since https://github.com/odoo/odoo/commit/87ffe5983be7f0f4a926b458c4bd1f13001048ec

victoralmau opened the pull request:
Prevent `price_unit` recursion error since https://github.com/odoo/odoo/commit/87ffe5983be7f0f4a926b458c4bd1f13001048ec

Please @pedrobaeza and @chienandalu can you review it?

@Tecnativa

victoralmau committed:
[FIX] purchase_discount: Prevent price_unit recursion error since https://github.com/odoo/odoo/commit/87ffe5983be7f0f4a926b458c4bd1f13001048ec

chienandalu commented on purchase_discount/models/purchase_order.py:
```diff
@@ -61,20 +61,8 @@ def _get_stock_move_price_unit(self):
-        if price_unit:
-            self.price_unit = price_unit
-        return price
+        return self._get_discounted_price_unit()
```
Only possible when this gets merged: https://github.com/odoo/odoo/pull/92933/files

Otherwise, you should provide a context to avoid recurssion

victoralmau replied:
Ok, I've added a context to avoid recursion.

legalsylvain reviewed (approved):
lgtm. Code review / no test.

thanks.

chienandalu commented on purchase_discount/models/purchase_order.py:
```diff
@@ -67,13 +67,15 @@ def _get_stock_move_price_unit(self):
         price = super()._get_stock_move_price_unit()
-        if price_unit:
-            self.price_unit = price_unit
+        if price_unit and not self.env.context.get("bypass_override_price_unit"):
```
This context check could be on the beggining of the method override and simply return super() if present

antoniocanovas reviewed (approved):
Functional review Ok !!

victoralmau commented:
Tests fail due to `purchase_propagate_qty`

pedrobaeza commented:
Please dig a bit on the CI error, although not related.

clementmbr reviewed (approved):
Thanks!

victoralmau commented:
> Please dig a bit on the CI error, although not related.

Reviewing the `purchase_propagate_qty` code and tests, I see that the behavior has changed from https://github.com/odoo/odoo/commit/87ffe5983be7f0f4a926b458c4bd1f13001048ec
and I'm not sure what is the correct behavior now.

pedrobaeza commented:
@rousseldenis there's a problem with `purchase_propagate_qty`that is caused by the same change in Odoo that this PR is fixing in other module.

pedroguirao commented:
LG! functional review OK

pedrobaeza committed:
[FIX] purchase_propagate_qty: Ci error due to upstream change

Since https://github.com/odoo/odoo/commit/ba4111e7a9202d801e3d51b55cad9659877e514a,
tests are failing.

Analyzing the roout cause, this module was testing things that are not
done by it, but a side effect of the previous status quo.

As such, we remove the tests that were testing this expected side
effects, as well as removing the reference in the README. If the
behavior wants to be restored, it should be explicitly handled in a
module with its own tests.

anajuaristi commented:
Functional :+1: LGTM

pedrobaeza commented:
Merging finally:

/ocabot merge patch
